Release `logzio k8s telemetry` v5.4.0

ralongit opened the pull request:
## Description 

-   Add SignalFx metrics receiver
-   Add Carbon metrics receiver
-   Add custom logs endpoint

<!-- 
    "This PR [adds/removes/fixes/replaces] the [feature/bug/etc].."
    Do not leave this blank.
    If relevant, describe the previous behaviour compared to the behaviour post your code change.
-->

## What type of PR is this?
#### (check all applicable)
- [x] 🍕 Feature 
- [ ] 🐛 Bug Fix
- [ ] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [x] ✅ Test
- [x] 🤖 Build / CI
- [ ] ⏩ Revert

## Added tests?

- [x] 👍 yes
- [ ] 🙅 no, because they aren't needed
- [ ] 🙋 no, because I need help from somebody

ralongit committed:
Feature signal fx metrics (#636)

* Initial commit - add signalFx metric

* remove signalfx logging config - will move to logs-collector

* add global enable flag for sfx

* improve conditions

ralongit committed:
Feature signal fx metrics (#636)

* Initial commit - add signalFx metric

* remove signalfx logging config - will move to logs-collector

* add global enable flag for sfx

* improve conditions

ralongit committed:
remove signal fx logging from telemetry chart

ralongit committed:
remove signal fx logging from telemetry chart

ralongit committed:
Add carbon flag (#637)

ralongit committed:
Add carbon flag (#637)

ralongit committed:
Update changelog and readme files

ralongit committed:
Update changelog and readme files

ralongit committed:
version 5.4.0

ralongit committed:
version 5.4.0

ralongit committed:
Remove global carbon value and add tests

ralongit committed:
Remove global carbon value and add tests

ralongit committed:
fix(ci): resolve test-protocol-receivers workflow failure for daemonset mode

- test-protocol-receivers workflow job
  - Added fail-fast: false - All matrix combinations now run independently instead of canceling when one fails
  - Fixed resource verification logic - Added conditional check to verify deployment for standalone mode and daemonset for daemonset mode
  - Updated deprecated ::set-output - Replaced with $GITHUB_OUTPUT environment file to avoid deprecation warnings

ralongit committed:
fix(ci): resolve test-protocol-receivers workflow failure for daemonset mode

- test-protocol-receivers workflow job
  - Added fail-fast: false - All matrix combinations now run independently instead of canceling when one fails
  - Fixed resource verification logic - Added conditional check to verify deployment for standalone mode and daemonset for daemonset mode
  - Updated deprecated ::set-output - Replaced with $GITHUB_OUTPUT environment file to avoid deprecation warnings

ralongit committed:
fix(test): correct service endpoints in metric generators

- Update signalfx-gen.yaml to use logzio-k8s-telemetry-otel-collector instead of *-standalone
- Update carbon-gen.yaml to use logzio-k8s-telemetry-otel-collector instead of *-standalone
- Resolves generator job timeouts in daemonset mode

ralongit committed:
fix(test): correct service endpoints in metric generators

- Update signalfx-gen.yaml to use logzio-k8s-telemetry-otel-collector instead of *-standalone
- Update carbon-gen.yaml to use logzio-k8s-telemetry-otel-collector instead of *-standalone
- Resolves generator job timeouts in daemonset mode

ralongit committed:
Add Debug Service and Pod Status to protocol test

ralongit committed:
Add Debug Service and Pod Status to protocol test

ralongit committed:
Add debug daemonset to the protocol tests job

ralongit committed:
Add debug daemonset to the protocol tests job

ralongit committed:
Add nodes troubleshooting for protocol job

ralongit committed:
Add nodes troubleshooting for protocol job

ralongit committed:
fix ports container configuration

ralongit committed:
fix ports container configuration

ralongit committed:
remove toleratioms from protocol tests

ralongit committed:
remove toleratioms from protocol tests

ralongit committed:
increase timeout for job

ralongit committed:
increase timeout for job

ralongit committed:
update collector endpoint

ralongit committed:
update collector endpoint

ralongit committed:
Fix signal fx tests

ralongit committed:
Fix signal fx tests

ralongit committed:
add tolerations to test protocol job

ralongit committed:
add tolerations to test protocol job

ralongit committed:
increase test protocol job generator timeout

ralongit committed:
increase test protocol job generator timeout

ralongit committed:
lower kind cluster version to match logs collector ds

ralongit committed:
lower kind cluster version to match logs collector ds

ralongit committed:
update protocol gens job to deployment

ralongit committed:
update protocol gens job to deployment

ralongit committed:
remove commented tolerations

ralongit committed:
remove commented tolerations

ralongit committed:
add collector logs

ralongit committed:
add collector logs

ralongit committed:
attempt adding tolerations

ralongit committed:
attempt adding tolerations

ralongit committed:
remove tolerations and lower k8s version

ralongit committed:
remove tolerations and lower k8s version

ralongit committed:
remove global tolerations from test-helm, remove space from protcol job

ralongit committed:
remove global tolerations from test-helm, remove space from protcol job

ralongit committed:
export env id

ralongit committed:
export env id

ralongit committed:
update signalfx test to python

ralongit committed:
update signalfx test to python

ralongit committed:
wait for nodes

ralongit committed:
wait for nodes

ralongit committed:
remove tolerations from protocol test

ralongit committed:
remove tolerations from protocol test

ralongit committed:
update signalfx gen to single container

ralongit committed:
update signalfx gen to single container

ralongit committed:
update signalfx metric label test

ralongit committed:
update signalfx metric label test

ralongit committed:
update carbon tests

ralongit committed:
update carbon tests

ralongit committed:
update carbon service endpoint

ralongit committed:
update carbon service endpoint

ralongit committed:
remove debug step and rename tests

ralongit committed:
remove debug step and rename tests

ralongit committed:
Update env id

ralongit committed:
Update env id

ralongit committed:
add setenv

ralongit committed:
add setenv

yotamloe commented on tests/resources/carbon-metrics-gen.yaml:
```diff
@@ -0,0 +1,108 @@
+                      timestamp = int(time.time())
+                      
+                      # Send the Carbon metric
+                      metric_data = f"test.carbon.metric;env_id={env_id};source=carbon-gen 1 {timestamp}"
```
This is graphite tag format, not sure if we want to test this format or dot notation format (no tags)

ralongit committed:
Release `logzio logs collector` v2.2.1 (#639)

* Add SignalFx logs

* Update CHANGELOG.md

* fix deployment and daemonset names

* fail-fast: false & revert cm filter tests

* update tests query

* fix signalfx test

* update dimension

* simplify query

* export env_id and update log type

* update env

* Update chart version & fix env id

* remove extra spaces

* Update tests env id

* debug signalfx tests

* Update signalfx_logs_e2e_test.go

* Fix missing component label in deployment

* Remove irrelevant signalfx param

* updates to config & tests

- Remove global.signalFx
- Simplify tests
- remove batch signalfx config, use base config
- add k8sattributes
- add log level
- add log type

ralongit committed:
Remove global flag & add dot notations support

ralongit committed:
Remove global flag & add dot notations support

ralongit committed:
Merge branch 'release/logzio-k8s-telemetry-v5.4.0' of https://github.com/logzio/logzio-helm into release/logzio-k8s-telemetry-v5.4.0

ralongit committed:
update protocol helm command syntax

ralongit committed:
add debug, otel stopped running

ralongit committed:
Update parser type to regex

ralongit committed:
attempt without parser
